bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -144,6 +146,26 @@ private boolean isAppInForeground() {
         return mActivityCount > 0 || mIsConfigChange;
     }
 
+    /**
+     * If it's a fresh start or back to main activity, we can safely clear cache
+     */
+    private boolean isFreshStart(@NonNull Activity activity, Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            return false;
+        }
+        if (!mIsFirstCreateCall) {
+            return false;
+        }
```

Comment:
I think we still need to be a bit more careful with how `mIsFirstCreateCall` is dealt with. For one, we should be setting the value inside this function. I also think it should be the first thing checked, so that we'd have:

```java
private boolean isFreshStart(@NonNull Activity activity, Bundle savedInstanceState) {
        if (!mIsFirstCreateCall) {
            return false;
        }
        mIsFirstCreateCall = false;
        if (savedInstanceState != null) {
            return false;
        }
        ...        
}
```